### PR TITLE
Fix semicolon parsing for async query

### PR DESCRIPTION
**Pull request by seankao-az**
### Description
Fix issue of no streaming job when `CREATE MATERIALIZED VIEW` statement ends with a semicolon.
Align parsing with upstream `opensearch-spark` repo:
https://github.com/opensearch-project/opensearch-spark/blob/b5ab7bd3ed135f1827df2cb0ca17422beb776985/flint-spark-integration/src/main/scala/org/opensearch/flint/spark/sql/FlintSparkSqlParser.scala#L58
 
### Issues Resolved
* #2630 
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by YANG-DB — DISMISSED** [review]

**Review by noCharger — APPROVED** [review]
Thanks for the change!

**Review by dai-chen — APPROVED** [review]
Thanks for the fix!

File: spark/src/main/antlr/SqlBaseParser.g4 [hunk 1]
```diff
@@ -480,7 +480,7 @@ dmlStatementNoWith
     | fromClause multiInsertQueryBody+                                             #multiInsertQuery
     | DELETE FROM identifierReference tableAlias whereClause?                      #deleteFromTable
     | UPDATE identifierReference tableAlias setClause whereClause?                 #updateTable
-    | MERGE INTO target=identifierReference targetAlias=tableAlias
+    | MERGE (WITH SCHEMA EVOLUTION)? INTO target=identifierReference targetAlias=tableAlias
```
Comment (dai-chen) [inline spark/src/main/antlr/SqlBaseParser.g4:483]: This is latest changes from Spark 3.3.1 grammar?

Comment (vamsi-amazon) [reply spark/src/main/antlr/SqlBaseParser.g4:483]: yes

**Review by vamsi-amazon — APPROVED** [review]

**Pull request merged**